docs/reference/query-dsl/random-sample-query.asciidoc

```diff
@@ -0,0 +1,62 @@
+[[query-dsl-random-sample-query]]
+=== Random Sample Query
+
+This query matches documents randomly according to the user-provided probability.
+For example, if there are 100 documents in the index and `probability = 0.5`,
+roughly 50 of the documents will be matched.
+
+[source,js]
+--------------------------------------------------
+GET /_search
+{
+    "query": {
+        "random_sample" : {
+            "probability" : 0.5
+        }
+    }
+}
+--------------------------------------------------
+// CONSOLE
+
+The `probability` parameter is required, and accepts values ` 0.0 > p > 1.0`.
+
+==== Seeding
+
+Optionally, the random number generator can be seeded with a Long.  This allows
+deterministic operation of the query so that the same documents are matched each
+time.
+
+This can be accomplished by adding the `seed` parameter:
```

Comment:
maybe the seed is a bit trappy since it only reproduces matches on a per point-in-time view of a shard. So if you go to a replica or refresh, you could match different documents?